packages/core/src/test/amazonqFeatureDev/util/files.test.ts

```diff
@@ -28,5 +30,32 @@ describe('file utils', () => {
             assert.strictEqual(result.zipFileChecksum.length, 44)
             assert.strictEqual(telemetry.repositorySize, 24)
         })
+
+        // Test the logic that allows the customer to modify root source folder
+        it('throws a ToolkitError with ContentLengthError code', async function () {
+            const fileNamePrefix = 'file'
+            const testContent = 'test content'
+            let files = 0
+
+            let fileContent = ''
+            while (Buffer.byteLength(fileContent) < maxFileSizeBytes - testContent.length) {
+                fileContent += testContent
+            }
+
+            for (let accumulatedSize = 0; accumulatedSize <= maxRepoSizeBytes; files++) {
+                accumulatedSize += Buffer.byteLength(fileContent)
+            }
+
+            const workspace = await createTestWorkspace(files, { fileNamePrefix, fileContent })
+            const telemetry = new TelemetryHelper()
+
+            try {
+                await prepareRepoData([workspace.uri.fsPath], [workspace], telemetry, {
+                    record: () => {},
+                } as unknown as Metric<AmazonqCreateUpload>)
+            } catch (err) {
+                assert.strictEqual(err instanceof ContentLengthError, true)
+            }
+        }).timeout(120 * 1000) // Allow test to run longer than 30 seconds
```

Comment:
Why does the test need to run longer than 30 seconds?

Reply:
Becuase I'm trying to create files that will break the 200MB limit for repo size, this file creation and all takes more time than 30s

Reply:
I'm adding the test specifically to avoid any future regressions as a refactor made by another team broke our feature.

Reply:
This seems like a convoluted way of creating a large set of files. Can't you just try something like?
```
fs.writeFile('file', new Buffer(<some size>));
```
That way you can just make one big file, and it might be faster.

If so, make sure to use the fs implementation if `srcShared/fs.ts`

Reply:
Alternatively, maybe we can stub `vscode.workspace.fs.stat(file.fileUri))` or something. I think that would be better than actually allocating a bunch of space on the testing machine (which would be our machines much of the time)

Reply:
> That way you can just make one big file, and it might be faster. Would like to avoid extending timeouts since tests already take a long time to run.

Yes, absolutely. In fact, this should be in `testInteg/`, that is where slow tests live.

Reply:
I've resorted to stubbing the `fsCommon.stat` function. As there is no need in that case to make the test longer, by just stubbing the function I can return a value that will make the logic throw the error.